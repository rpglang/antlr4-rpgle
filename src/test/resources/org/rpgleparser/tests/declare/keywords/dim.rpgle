     D xxRecipientArr  S                   LIKE(xpRecipientArr)
     D                                     DIM(RecipientArrElements)