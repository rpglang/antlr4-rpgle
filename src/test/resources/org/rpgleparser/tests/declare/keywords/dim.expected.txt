/*===TOKENS===*/
DS_FIXED                 <D>
NAME                     <xxRecipientArr>
EXTERNAL_DESCRIPTION     < >
DATA_STRUCTURE_TYPE      < >
DEF_TYPE_S               <S >
FROM_POSITION            <       >
TO_POSITION              <       >
DATA_TYPE                < >
DECIMAL_POSITIONS        <  >
RESERVED                 < >
KEYWORD_LIKE             <LIKE>
'('                      <(>
ID                       <xpRecipientArr>
')'                      <)>
KEYWORD_DIM              <DIM>
'('                      <(>
ID                       <RecipientArrElements>
')'                      <)>
/*===TREE===*/
(r
  (statement
    (dspec
      D
      (ds_name xxRecipientArr)
      ' '
      ' '
      'S '
      '       '
      '       '
      ' '
      '  '
      ' '
      (keyword
        (keyword_like
          LIKE
          (
          (simpleExpression (identifier (free_identifier (idOrKeyword xpRecipientArr))))
          )
        )
      )
      (keyword
        (keyword_dim
          DIM
          (
          (simpleExpression
            (identifier (free_identifier (idOrKeyword RecipientArrElements)))
          )
          )
        )
      )
      <EOF>
    )
  )
)
/*======*/